.gitlab/source_test/golang_deps_generate.yml

```diff
@@ -0,0 +1,22 @@
+---
+# golang_deps_generate stage
+# Contains the step to build the golang dependency chain and preserve it
+golang_deps_generate:
+  rules:
+    !reference [.on_main_or_release_branch]
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  needs: ["linux_x64_go_deps"]
+  variables:
+    ARCH: "x64"
+  before_script:
+    - !reference [.retrieve_linux_go_deps]
+    - python3 -m pip install --upgrade --ignore-installed pip setuptools
```

Comment:
We don't usually need to do `--ignored-installed`, why do we need it here?

Reply:
Copy/paste from https://github.com/DataDog/datadog-agent/blob/main/.gitlab/source_test/linux.yml#L128. I don't know for sure if it's needed though.